src/rgw/rgw_auth_s3.cc

```diff
@@ -700,13 +703,18 @@ std::string get_v4_string_to_sign(CephContext* const cct,
                                   const sha256_digest_t& canonreq_hash)
 {
   const auto hexed_cr_hash = buf_to_hex(canonreq_hash);
-  const auto string_to_sign = std::string()
+
+  /* We aren't adding the extra byte for the terminating 0 as it would be
+   * annihilated by the "-1" coming from hexed_cr_hash. */
```

Comment:
note that `string::reserve(n)` guarantees room for n characters. the implementation will add 1 internally for the null terminator
```c++
  std::string a;
  a.reserve(36);
  std::cout << a << '[' << a.capacity() << ']' << std::endl;
  a.append(36, 'a');
  std::cout << a << '[' << a.capacity() << ']' << std::endl;
```
produces:
> [36]
aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa[36]